Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/Game.java

```diff
@@ -0,0 +1,115 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.util.ArrayList;
+
+public class Game {
+	protected 	Dictionary 	dictionary;		
+	protected 	Player 		player;		  	
+	protected 	char[] 		etalon;			
+	protected	ArrayList 	<Character> 	guessedLetters;
+	protected	char[] 		openedWord;		
+	protected	ArrayList	<Character>		wrongLetters;
+	protected 	UserInterface ui;			
+	protected 	final int 	TOTAL_GUESSES=9;
+	protected 	int 		currentGuesses;
+
+
+	public Game(){
+		dictionary = new ArrayDictionary ("words.txt");
+		ui = new GUIMainWindow(this);
+		player = new SimplePlayer(ui.playerName());
```

Comment:
Again, why does the UI get to decide what sort of player to use? Its kinda pointless having lots of different types of users anyway if the Game logic decides anyway, and callers can't specify what they want.